### PR TITLE
Currently there is no limit on number of records getting deleted by cleanup script. Adding limit to make sure database is not affected by cleanup script execution.

### PR DESCRIPTION
**Pull request by jakhani**
When cleanup script was executed for the first time, it has deleted so many records that it affected slave server. I am adding limit to make sure that no matter how frequently we run cleanup script, it only deletes limited records and slave server does not get affected.

### CODE REVIEW

File: azkaban-web-server/src/main/java/azkaban/webapp/ExecutionLogsCleaner.java [hunk 1]
```diff
@@ -25,10 +25,14 @@
    private static final long DEFAULT_EXECUTION_LOGS_RETENTION_MS = 3 * 4 * 7
        * 24 * 60 * 60 * 1000L;
 
-   // 1 hour
-   private static final long DEFAULT_LOG_CLEANUP_INTERVAL_SECONDS = 60 * 60;
+   // 1 minute
+   private static final long DEFAULT_LOG_CLEANUP_INTERVAL_SECONDS = 60;
```
Comment (li-pkhandelwal) [inline azkaban-web-server/src/main/java/azkaban/webapp/ExecutionLogsCleaner.java:29]: cleaning up every minute for something which has retention of 12 weeks seems too frequent. Do we know how long does it take on avg to execute this statement for 1000 deletions, and do we have over 1000 executions for clusters in a minute. 
This is not cleaning up everything too, we could probably run this clean up every hour with some batch_size in a loop while there are records to delete and committing each batch.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java [hunk 2]
```diff
@@ -47,6 +43,10 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static azkaban.utils.ThinArchiveUtils.*;
```
Comment (abhisheknath2011) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java]: Looks like the the * imports got introduced by IDE or by formatting. Might want to change these to individual imports.

File: azkaban-web-server/src/main/java/azkaban/webapp/ExecutionLogsCleaner.java (hunk 1 above)
Comment (djaiswal83) [reply azkaban-web-server/src/main/java/azkaban/webapp/ExecutionLogsCleaner.java:29]: 48K log entries seems too less for clusters like UMP where at any given time 500-1500 flows are running with each flow running multiple jobs. However, I agree 1 minute is too aggressive. Lets find out how many logs are added in 24 hrs in UMP and then work on the frequency backwards.

Comment (jakhani) [reply azkaban-web-server/src/main/java/azkaban/webapp/ExecutionLogsCleaner.java:29]: I will keep 1 hour as default. With that, we can set this differently based on load on cluster by setting config value for this frequency.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java (hunk 2 above)
Comment (jakhani) [reply azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java]: This file change should not have been committed. I have reverted it.

File: azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java [hunk 3]
```diff
@@ -135,12 +135,12 @@ private void uploadLogFile(final DatabaseTransOperator transOperator, final int
     }
   }
 
-  int removeExecutionLogsByTime(final long millis)
+  int removeExecutionLogsByTime(final long millis, final int recordCleanupLimit)
       throws ExecutorManagerException {
     final String DELETE_BY_TIME =
-        "DELETE FROM execution_logs WHERE upload_time < ?";
+        "DELETE FROM execution_logs WHERE upload_time < ? LIMIT ?";
```
Comment (li-pkhandelwal) [inline azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java:141]: This is not maintaining retention time for logs as we are deleting only 1000 entries and we might end up with huge number of rows spanning over multiple months over a period of time if the cluster generates more rows as we are restricting ourselves to delete only 24k rows/day. 
Like I pointed out earlier a better would be to probably run this clean up every hour with some batch_size in a loop while there are records to be deleted(i.e older than 12 weeks) and committing each batch.

Comment (li-pkhandelwal) [reply azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java:141]: I understand this limit is in config, but deleting fixed number of rows is not deterministic for enforcing retention period

**Pull request closed without merging**